src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/UpdateSessionTemplateValidator.kt

```diff
@@ -64,9 +65,10 @@ class UpdateSessionTemplateValidator(
       if (newValidToDate != existingValidToDate) {
         // if the new validToDate is not null or before existing validToDate
         if ((newValidToDate != null && existingValidToDate == null) || (newValidToDate != null && newValidToDate.isBefore(existingValidToDate))) {
-          // check if there are any visits (any visit status) after the new valid to date
-          if (visitRepository.hasVisitsForSessionTemplate(existingSessionTemplate.reference, newValidToDate.plusDays(1))) {
-            return "Cannot update session valid to date to $newValidToDate for session template - ${existingSessionTemplate.reference} as there are visits associated with this session template after $newValidToDate."
+          // check if there are any booked or reserved visits (any visit status) after the new valid to date
+          val visitStatues = listOf<VisitStatus>(VisitStatus.BOOKED, VisitStatus.RESERVED, VisitStatus.CHANGING)
+          if (visitRepository.hasVisitsForSessionTemplate(existingSessionTemplate.reference, newValidToDate.plusDays(1), visitStatues)) {
```

Comment:
would make this a constant or put it directly in the query, could also put it in it's own local method hasVisitsForSessionTemplate

Reply:
should I make 2 methods? for some instances we might need Cancel in some cases e.g. if they are trying to change the from date but any type of visits exists

Reply:
If we need 2 now, sure!